### PR TITLE
man: Add new Microsoft CAs to example

### PR DESCRIPTION
**Pull request by medhefgo**
Also, let's use the links from the Microsoft documentation as they are
guaranteed to be stable.

Fixes: #29102

### CODE REVIEW

**Comment by medhefgo** [comment]
@kerneis-anssi Please have a look. The way I read their docs, both 2011 and 2023 keys should be used, right?

**Review by bluca — APPROVED** [review]
Yes, you are supposed to have both CAs available, at least until the old ones expire, possibly later too.

**Comment by medhefgo** [comment]
> Yes, you are supposed to have both CAs available, at least until the old ones expire, possibly later too.

I'm pretty sure the UEFI spec says it does not do expiration date…

**Comment by kerneis-anssi** [comment]
> The way I read their docs, both 2011 and 2023 keys should be used, right?

That's correct.

Overall, the PR looks good to me. There may be room for clarification in the file extensions, see this SO answer for background information: https://serverfault.com/a/9717

TL;DR:
- MS `.crt` files are in fact `.der` files (at least for the files you are downloading here).
- MS `.base64` files are probably intended to be `.pem` files except they forgot  to add `-----BEGIN CERTIFICATE-----` so it breaks the RFC (and every tool using openssl that would try and parse them).
- `sbsiglist` can only handle `.der` files, not `.pem` ones (unlike `cert-to-efi-sig-list`).

Since the script generates explicitly `.der` and `.pem` files for our custom hierarchy above, I suggest using `.der` instead of `.crt` for MS files, so it's obvious what that they are in the same format. It seems apt to keep the `.base64` option (good catch!) because that's really what those half-baked bags of bytes are.

**Comment by bluca** [comment]
> > Yes, you are supposed to have both CAs available, at least until the old ones expire, possibly later too.
> 
> I'm pretty sure the UEFI spec says it does not do cert expiration dates…

Indeed and firmwares don't care either typically, but just talking about the current advice we give to OEMs. It's "both" for the foreseeable future, although it might change after they expire, there's no decision/clarity yet.

**Comment by kerneis-anssi** [comment]
And likewise for consistency, either rename .crt on line 311 above to .pem, or directly emit a .der file when generating the file with `-outform der`.

**Comment by medhefgo** [comment]
> It seems apt to keep the `.base64` option (good catch!) because that's really what those half-baked bags of bytes are.

Credit goes to https://github.com/Foxboron/sbctl/issues/235. Though, it sounds like either of you might be in a position to get that fixed?

> And likewise for consistency, either rename .crt on line 311 above to .pem, or directly emit a .der file when generating the file with `-outform der`.

sbvarsign only likes `der` for whatever reason. :(

**Review by kerneis-anssi — APPROVED** [review]

**Review by bluca — APPROVED** [review]
